pkg/apis/core/v1alpha1/types_utils.go

```diff
@@ -41,13 +41,16 @@ type Condition struct {
 	LastTransitionTime metav1.Time `json:"lastTransitionTime" protobuf:"bytes,3,opt,name=lastTransitionTime"`
 	// Last time the condition was updated.
 	LastUpdateTime metav1.Time `json:"lastUpdateTime" protobuf:"bytes,4,opt,name=lastUpdateTime"`
+	// Last time the health check was done.
+	// +optional
+	LastHeartbeatTime *metav1.Time `json:"lastHeartbeatTime,omitempty" protobuf:"bytes,5,opt,name=lastHeartbeatTime"`
 	// The reason for the condition's last transition.
-	Reason string `json:"reason" protobuf:"bytes,5,opt,name=reason"`
+	Reason string `json:"reason" protobuf:"bytes,6,opt,name=reason"`
```

Comment:
Protobuf index change is not backward compatible change.
```suggestion
	Reason string `json:"reason" protobuf:"bytes,5,opt,name=reason"`
```

Reply:
So `LastHeartbeatTime` should be the last index (ie; 8) here?